### ISSUE TITLE
Trainable parameters in TF Hub release

### ISSUE THREAD

**Issue by martiansideofthemoon** [allenai/bilm-tf#66]
I had a few questions about the set of trainable parameters in the TF Hub releases of ELMo. The [initial release](https://www.tensorflow.org/hub/modules/google/elmo/1) mentions that the LSTM cell parameters are `trainable` (and this is what I expected, fine-tuning on the downstream task's supervised labels). However, I recently came across [this paper](https://arxiv.org/pdf/1801.06146.pdf) which mentioned that the LSTM parameters in ELMo are fixed, and it also seems to be the case in the [current release](https://www.tensorflow.org/hub/modules/google/elmo/2) of ELMo on TF Hub.

1. Were the LSTM parameters kept fixed during the experiments described in the paper? (and the only fine-tuning done ignored the supervised labels, and used the training set of the downstream task for language modelling?) 
2. Did you notice significant performance drops keeping the LSTM cell parameters `trainable` during the downstream task?

**Comment by DeNeutoy** [comment]
Hi there,

1. LSTM parameters were kept fixed. For each downstream task, we trained a scalar weighted mixture of the layers (i.e a single scalar trainable parameter per layer).

2. Yes, fine tuning Language Models is a difficult, open problem. It is the subject of considerable current research, e.g https://arxiv.org/abs/1801.06146 and https://s3-us-west-2.amazonaws.com/openai-assets/research-covers/language-unsupervised/language_understanding_paper.pdf

Also, note that we did not implement the tensorflow hub ELMo model.
